tests/run-programs.py

```diff
@@ -54,7 +54,7 @@
     output += completed_process.stdout.decode('UTF-8')
     end = time.perf_counter()
 
-    if 'FAIL: ' not in output or 'cost exceeded' not in output:
+    if 'FAIL: ' not in output or ('cost exceeded' not in output and 'too many pairs' not in output):
```

Comment:
some of these stress-tests exceed the allocation limit now, and they can fail in two waus